sematic/plugins/external_resource/ray/cluster.py

```diff
@@ -0,0 +1,387 @@
+# Standard Library
```

Comment:
LOTS of error handling in here. Want to turn things into user/us-understandable messages that will show up in our UI for the run whenever possible. Recall that a lot of this code executes on the server, so getting it to the UI means going through the status message field of the resource.